server/src/main/java/org/opensearch/common/cache/store/config/StoreAwareCacheConfig.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.store.config;
+
+import org.opensearch.common.cache.store.listeners.StoreAwareCacheEventListener;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+
+import java.util.function.ToLongBiFunction;
+
+/**
+ * Configurations related to store aware caches
+ *
+ * @opensearch.internal
+ */
+public class StoreAwareCacheConfig<K, V> {
```

Comment:
This class living in core means that core needs to anticipate all configuration parameters that any cache would potentially want.

Suppose we add a remote cache hosted on a cloud service, and connecting to that service endpoint requires a URL, a service region identifier,  parameters to talk to a credential provider,  etc. Would we add all of those parameters to this config class?

Alternatively, core can read the cache configuration as a JSON object (essentially just a `Map<String, Object>`, where nested objects are also `Map<String, Object>`). Then it's up to the cache's factory to interpret that map correctly. 

Core just needs to look up the correct cache factory.

Reply:
Makes sense. I am now following a setting based model where each plugin will have its own setting to create a object. I am keeping this class to only have some common configurations.